packages/contracts/src/forwarder/Forwarder.sol

```diff
@@ -152,7 +154,8 @@ contract Forwarder is IForwarder, ERC165 {
                 "\x19\x01", domainSeparator,
                 keccak256(_getEncoded(req, requestTypeHash, suffixData))
             ));
-        require(digest.recover(sig) == req.from, "FWD: signature mismatch");
+        // solhint-disable-next-line avoid-tx-origin
+        require(tx.origin == DRY_RUN_ADDRESS || digest.recover(sig) == req.from, "FWD: signature mismatch");
```

Comment:
please reorder them, adding `|| tx.origin at the end`
You're preventing any possible estimation of gas, even of the internal calls.

Reply:
`digest.recover(sig)` will revert if `sig` is not a valid ecdsa signature, and I don't want to put a stub byte array in `sig` to reorder things. Can you please explain how reordering things will help with gas estimation? I guess you can estimate all you want as long as `tx.origin != DRY_RUN_ADDRESS`, can't you?

Reply:
the `tx.origin== DRY_RUN_ADDRESS` check is very cheap - about 10 gas - and placing it first means it "short-circuit" and doesn't calculate the rest of the line - which means dry-run will cost ~3000 less.
Putting that check at the end means the gas "fluctuation" with and without dry-run is roughly 10 gas.
(in normal run, the sig is checked and the "`tx.origin==DRU_RUN_ADDRESS`" check is skipped.
in dry-run, the signature is checked (and fail, and the same cost), and then the dry-run check is executed

Reply:
I really fail to understand where does your interest in gas consumption in dry run come from. I think it is completely irrelevant.